### ISSUE TITLE
rsample() in distributions.relaxed_bernoulli

### ISSUE THREAD

**Issue by eelxpeng** [pytorch/pytorch#6700]
I noted some discrepency of the implementation of `LogitRelaxedBernoulli ` in distributions.relaxed_bernoulli with the following cited paper in the code

> [1] The Concrete Distribution: A Continuous Relaxation of Discrete Random Variables
>     (Maddison et al, 2017)

The `rsample()` function is implemented as 

`(uniforms.log() - (-uniforms).log1p() + probs.log() - (-probs).log1p()) / self.temperature`

But according to the paper `(uniforms.log() - (-uniforms).log1p() + probs.log() ) / self.temperature` is what should be done. There is no `log(1-alpha)` in the original paper. I know that this is also how RelaxedBernoulli is implemented in the Tensorflow Distribution package. I wonder if this is a bug instead.

**Comment by apaszke** [comment]
cc: @fritzo

**Comment by fritzo** [comment]
cc @rachtsingh 

@eelxpeng can you point to which definition in the paper you're translating? It looks like [Proposition 2 part (a)](https://arxiv.org/pdf/1611.00712.pdf) is correctly implemented by `RelaxedBernoulli`, where `alpha` is a positive real location parameter (in odds units) and:
```py
log(alpha) == probs.log() - (-probs).log1p()
```
Or am I missing something?

**Comment by eelxpeng** [comment]
@fritzo sorry, that's my mistake. I thought `alpha = probs`, but it turns out that `alpha = probs/(1-probs)`.

**Comment by rachtsingh** [comment]
Ah yeah, I too was confused about the parametrization the first time I implemented this. Thanks for checking!

**Comment by HanGuo97** [comment]
Out of curiosity, what's the reason for first transforming `logits` into `probs` and do `probs.log() - (-probs).log1p())`? This seems rather unnecessary as the latter term simply equals `logits`.

**Comment by fritzo** [comment]
@HanGuo97 this may merely be a missed optimization; feel free to submit a PR.

**Comment by HanGuo97** [comment]
Got it, thanks!
